src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -100,12 +103,33 @@ public void storeMessage(Message message) {
     // people may include implicit context/reference to previously sent messages,
     // and this order makes the chronology of the conversation and context
     // more intuitive and understandable in the messages list page. 
-    Query query =
-        new Query("Message")
-            .setFilter(directMessages)
-            .addSort("timestamp", SortDirection.ASCENDING);
+    Query query = hasUserRestriction(loggedInUser, otherUser)
```

Comment:
Now that there is a `getAllMessages()` function, is there a need to have this conditional logic?

Instead, I would recommend inserting:

```
   Preconditions.checkArgument(!Strings.isNullOrEmpty(loggedInUser), "loggedInUser is null or empty");
   Preconditions.checkArgument(!Strings.isNullOrEmpty(otherUser), "otherUser is null or empty");
```

... at the very top of this function and not accepting this null/empty case.

Reply:
I did not know that ```Preconditions.checkArgument()``` was a thing. I will take a look and implement it.
Thank you for letting me know